src/platform/graphics/vertex-buffer.js

```diff
@@ -21,8 +21,11 @@ class VertexBuffer {
      * @param {number} numVertices - The number of vertices that this vertex buffer will hold.
      * @param {number} [usage] - The usage type of the vertex buffer (see BUFFER_*). Defaults to BUFFER_STATIC.
      * @param {ArrayBuffer} [initialData] - Initial data.
+     * @param {object} [options] - Object for passing optional arguments.
+     * @param {boolean} [options.storage] - Defines if the vertex buffer can be used as a storage
+     * buffer by a compute shader. Defaults to false. Only supported on WebGPU.
      */
-    constructor(graphicsDevice, format, numVertices, usage = BUFFER_STATIC, initialData) {
+    constructor(graphicsDevice, format, numVertices, usage = BUFFER_STATIC, initialData, options) {
```

Comment:
Hmm. This constructor signature is getting a bit messy. If you pass options, you need to set a value of `initialData`. which a lot of devs never pass. Feels like if this was written from scratch, `usage` and `initialData` would be in `options`. IIRC, `initialData` was added originally as a bit of a hack...

Reply:
I agree, but the devs can pass `uninitialised` for those to get the default values.
But maybe it's time we move last 3 or so parameters to options?

Reply:
Looking at this, I think a separate PR would be a good idea here too.